src/pke/include/scheme/ckksrns/ckksrns-fhe.h

```diff
@@ -115,9 +115,10 @@ class CKKSBootstrapPrecom {
 class FHECKKSRNS : public FHERNS {
     using ParmType = typename DCRTPoly::Params;
 
-    const double K_SPARSE  = 14.0;   // upper bound for the number of overflows in the sparse secret case
-    const double K_UNIFORM = 512.0;  // upper bound for the number of overflows in the uniform secret case
-    static const int32_t R = 6;      // number of double-angle iterations in CKKS bootsrtrapping
+    const double K_SPARSE     = 14.0;   // upper bound for the number of overflows in the sparse secret case
+    const double K_UNIFORM    = 512.0;  // upper bound for the number of overflows in the uniform secret case
+    static const int32_t R    = 6;      // number of double-angle iterations in CKKS bootsrtrapping
+    double m_correctionFactor = 0.0;    // correction factor, which we scale the message by to improve precision
```

Comment:
why do we have a global variable here? this is not a constant.

Reply:
I used a class variable because we input correctionFactor in EvalBootstrapSetup, but we use it in EvalBootstrap.